### PR TITLE
Get rid of copypaste in filter applied panels

### LINKED ISSUE #615: Get rid of copypaste in filter applied panels
In ProblemFilterApplied.tsx and in CandidateFilterApplied.tsx

### PR DESCRIPTION
**Pull request by nyet-ty**
Resolve #615 

## PR includes

-   [x] Refactor

### CODE REVIEW

**Review by tsumo — CHANGES_REQUESTED** [review]
Looks like reduce is the same everywhere. It could be moved to mapToAppliedString

File: src/modules/crewTypes.ts [hunk 1]
```diff
@@ -50,6 +50,7 @@ export interface Vacancy {
 
     activeSince?: Date;
     timeAtWork: number;
+    [key: string]: unknown;
```
Comment (tsumo) [inline src/modules/crewTypes.ts:53]: Why?

File: src/utils/index.ts [hunk 2]
```diff
@@ -124,3 +114,36 @@ export const standConfig = {
...
+}
+
+interface TEntity {
+    id: string | number;
+    name?: string | null;
+    email?: string;
+    [key: string]: unknown;
+}
+
+export function arrayToAppliedString(
```
Comment (tsumo) [inline src/utils/index.ts:130]: T extends TEntity

array: T[]

key: keyof T

**Review by tsumo — CHANGES_REQUESTED** [review]

File: src/modules/crewTypes.ts (hunk 1 above)
Comment (nyet-ty) [reply src/modules/crewTypes.ts:53]: without it typescript doesn't know that typeof keyof id === 'string'

File: src/utils/index.ts (hunk 2 above)
Comment (nyet-ty) [reply src/utils/index.ts:130]: This one fix your previous comment, thanks)

File: src/modules/crewTypes.ts (hunk 1 above)
Comment (tsumo) [reply src/modules/crewTypes.ts:53]: Can be solved with `extends string` and generics

**Review by tsumo — CHANGES_REQUESTED** [review]

**Review by tsumo — APPROVED** [review]

**Review by 9teen90nine — APPROVED** [review]

**Pull request merged**